track.py

```diff
@@ -61,12 +62,38 @@ class Track:
     @staticmethod
     def from_xml_file(filename):
         xmldoc = parse(filename)
+        width = get_track_root_info(xmldoc, 'width')
+        profil_steps_length = get_track_root_info(xmldoc, 'profil steps length')
+
         segments = []
         for section in get_main_track(xmldoc).childNodes:
             if section.nodeType == Node.ELEMENT_NODE:
-                segments.append(Segment.from_xml(section))
+                s = Segment.from_xml(section)
+                if s.end_radius is None:
+                    segments.append(s)
+                else:
+                    length = abs((s.radius + s.end_radius)/2.0 * s.arc)
+                    num_steps = int(length/4.0) + 1
```

Comment:
I don't understand this line. Shouldn't `num_steps` be `length/profil_steps_length`? The variable `profil_steps_length` is set but never used.